kernel/cbytegen.ml

```diff
@@ -535,12 +574,13 @@ let compile_fv_elem reloc fv sz cont =
 
 let rec compile_fv reloc l sz cont =
   match l with
-  | [] -> cont
+  | [] -> set_max_stack_size sz; cont
   | [fvn] -> compile_fv_elem reloc fvn sz cont
```

Comment:
Since there is a special case for the length-1 list, the length-0 list will almost never be triggered (and when it is, you don't need any extra stack space), so the code seems plain wrong here. It should have been something like

    | [] -> cont
    | [fvn] -> set_max_stack_size (sz + 1); compile_fv_elem reloc fvn sz cont

In case you wonder why I suggest `+ 1` even though there is no push in that branch, that is because there is an implicit push in all the opcodes that prepare a closure:

    if (nvars > 0) *--sp = accu;